firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/crcelements.txt

```diff
@@ -0,0 +1,212 @@
+      THEN("the crc accessor method returns 0 before the first write method is called")
+      THEN("The first write method call returns ok status")
+      THEN("After the first write method is called, the crc accessor method returns a value equal to the result of directly computing a CRC on the payload.")
+      THEN("The CRC field of the body's header matches the value returned by the crc accessor method.")
+      THEN("The body's payload section correctly stores the payload as '0x01 0x05 0x01 0x02 0x03 0x04 0x05'")
+      THEN("The output buffer is as expected")
+      # parse
+      THEN("The parse method reports ok status")
+      THEN("After the parse method is called, the value returned by the crc accessor method is equal to the crc field of the input_buffer body's header")
+      THEN("the payload returned from the payload accessor method is equal to the payload from the body")
+      THEN("the payload given in the ParsedCRCElement constructor is independent of the input_buffer body")
+      # write  
+      THEN("The second write method call returns ok status")
+      THEN("After the second write method is called, the crc accessor method returns a value equal to the result of directly computing a CRC on the payload.")
+      THEN("The CRC field of the body's header matches the value returned by the crc accessor method.")
+      THEN("The body's payload section correctly stores the payload as '0x01 0x05 0x01 0x02 0x03 0x04 0x05'")
+      THEN("The output buffer is as expected")
+
+/* CRCElementReceiver
+Scenario: CRCElementReceiver correctly parses CRCElement bodies and performs consistency checking on them
+
+  GIVEN("A CRC element receiver of capacity 254 bytes and an output_crcelement constructed with an empty payload buffer")
+    // assertions carried out in GIVEN:
+    // The output_crcelement's crc accessor method returns 0
+    // The payload buffer returned by the output_crcelement's payload accessor method is empty
+
+    WHEN("An empty input_buffer is given to the crc element receiver")
+      THEN("the transform method reports invalid parse status")
+      THEN("After the transform method is called, the value returned by the output_crcelement's crc accessor method remains unchanged")
+      THEN("The output_crcelement's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_crcelement's payload buffer is independent of the data in input_buffer")
+
+    WHEN("A body with less than 4 bytes is given to the crc element receiver")
+      // body of length 1,2,3 bytes is passed, passing 4 bytes will give a invalid_crc error instead
+      THEN("the transform method reports invalid parse status")
+      THEN("After the transform method is called, the value returned by the output_crcelement's crc accessor method remains unchanged")
+      THEN("The output_crcelement's payload accessor method returns a reference to the payload given in its constructor")
```

Comment:
What should the contents of the payload buffer be after parse method is called? (I think the contents should be unchanged)